lib/ansible/plugins/callback/__init__.py

```diff
@@ -95,6 +95,9 @@ def _dump_results(self, result, indent=None, sort_keys=True, keep_invocation=Fal
         if 'exception' in abridged_result:
             del abridged_result['exception']
 
+        # ensure we don't hit decode errors on displaying errors
+        abridged_result = to_text(abridged_result)
+
```

Comment:
If result is sent to to_text() before strip_internal_keys() is called, then the call to strip_internal_keys() results in ...

```
 [WARNING]: Failure using method (v2_runner_item_on_failed) in callback plugin
(<ansible.plugins.callback.default.CallbackModule object at 0x7f48f2306fd0>): 'unicode' object has no
attribute 'copy'
```